worker/caasadmission/k8s/controller.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package k8s
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/juju/errors"
+	"gopkg.in/juju/worker.v1/catacomb"
+
+	"github.com/juju/juju/caas/kubernetes/provider"
+)
+
+type Mux interface {
+	AddHandler(string, string, http.Handler) error
+	RemoveHandler(string, string)
+}
+
+type Logger interface {
+	Debugf(string, ...interface{})
+	Errorf(string, ...interface{})
+	Infof(string, ...interface{})
+}
+
+// Kubernetes controller responsible
+type Controller struct {
+	catacomb catacomb.Catacomb
+	logger   Logger
+}
+
+func AdmissionPathForModel(modelUUID string) string {
+	return fmt.Sprintf("/k8s/admission/%s", url.PathEscape(modelUUID))
+}
+
+func NewController(
+	logger Logger,
+	mux Mux,
+	path string,
+	admissionCreator AdmissionCreator,
+	rbacMapper provider.RBACMapper) (*Controller, error) {
+
+	c := &Controller{
+		logger: logger,
+	}
+
+	if err := catacomb.Invoke(catacomb.Plan{
+		Site: &c.catacomb,
+		Work: c.makeLoop(admissionCreator,
+			admissionHandler(logger, rbacMapper), mux, path),
+	}); err != nil {
+		return c, errors.Trace(err)
+	}
+
+	if err := c.catacomb.Add(rbacMapper); err != nil {
+		return c, errors.Trace(err)
+	}
```

Comment:
If we're using rbacMapper as an arg to Invoke, this should happen up above beforehand

Also, if the controller errors and is restarted, that will also (unnecessarily in most cases) restart the rbac mapper. I thought maybe we had discussed having the rbacmapper as a separate worker, plugged into this one as a dependency?

Reply:
Yeah we did. The plan is to do it. Will knock it out today, should be a simple move.